### PR TITLE
Add HPMC pair potential howto guide.

### LINKED ISSUE #1505: Add a quick starter guide for adding isotropic interactions in HPMC
## Tutorial description

This is inspired by the tutorial on how to model patchy particles with HPMC glotzerlab/hoomd-examples#84. 
The goal of this guide: demonstrate how to add isotropic interactions in HPMC. 
Like in glotzerlab/hoomd-examples#84, it would be helpful to mention the commonly used potentials, why one might want to use this, etc.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Add a guide that shows how to apply a hard sphere square well potential in HPMC.

## Motivation and context

See the discussion in https://github.com/glotzerlab/hoomd-examples/pull/84.

Resolves #1505 

## How has this been tested?

I executed the example and got the expected output.

## Change log

```
Added:

* Guide: How to apply arbitrary pair potentials in HPMC
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by Charlottez112** [comment]
Looks good to me. Thank you!

**Review by Charlottez112 — APPROVED** [review]

File: sphinx-doc/howto/cpppotential.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+3. Set the `pair_potential <hoomd.hpmc.integrate.HPMCIntegrator.pair_potential>` property of the
+   HPMC integrator.
+
+This code demonstrates the hard sphere square well potential.
+
+.. literalinclude:: cpppotential.py
+    :language: python
+
+Use HPMC with pair potentials when to model hard particle core interactions or discontinuous steps
+in the potential. Use Molecular dynamics for models with continuous potentials.
```
Comment (tommy-waltmann) [inline sphinx-doc/howto/cpppotential.rst]:
```suggestion
Use HPMC with pair potentials to model interactions with discontinuous steps
in the potential. Use molecular dynamics for models with continuous potentials.
```

File: sphinx-doc/howto/molecular.py [hunk 2]
```diff
@@ -1,27 +1,24 @@
-# Copyright (c) 2009-2023 The Regents of the University of Michigan.
-# Part of HOOMD-blue, released under the BSD 3-Clause License.
-
 import hoomd
 import gsd.hoomd
 
-snapshot = gsd.hoomd.Snapshot()
+frame = gsd.hoomd.Frame()
```
Comment (tommy-waltmann) [inline sphinx-doc/howto/molecular.py:4]: Should changes to be compatible with the upcoming GSD 3.0 release be on a different PR and/or delayed until after GSD 3.0 is released?

File: hoomd/hpmc/pair/user.py [hunk 3]
```diff
@@ -215,6 +215,9 @@ class CPPPotential(CPPPotentialBase):
         `CPPPotential` is **experimental** and subject to change in future minor
         releases.
 
+    Example:
+        See :doc:`howto/cpppotential`.
+
```
Comment (tommy-waltmann) [inline hoomd/hpmc/pair/user.py:220]: Should we also refer to the new howto guide in `CPPPotentialUnion`?

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
I have one suggested re-wording and two comments to think about.

File: sphinx-doc/howto/molecular.py (hunk 2 above)
Comment (joaander) [reply sphinx-doc/howto/molecular.py:4]: `Frame` is available now in GSD 2.8.0.

File: hoomd/hpmc/pair/user.py (hunk 3 above)
Comment (joaander) [reply hoomd/hpmc/pair/user.py:220]: done.

**Review by tommy-waltmann — APPROVED** [review]
LGTM!
